pyquil/api/_qvm.py

```diff
@@ -51,7 +50,12 @@ def check_qvm_version(version: str) -> None:
 @dataclass
 class QVMExecuteResponse:
     executable: Program
-    memory: Mapping[str, np.ndarray]
+    data: QVMResultData
+
+    @property
+    def memory(self) -> Dict[str, np.ndarray]:
+        register_map = self.data.to_register_map()
+        return {key: matrix.to_ndarray() for key, matrix in register_map.items()}
```

Comment:
Would the `to_ndarray()` call ever be expensive enough that it would be worth _us_ caching it automatically so the user does not have to?

Reply:
It's a good question and I had the same thought, but keeping it simple instead of optimizing prematurely. We will be doing v4 benchmarks, so if performance seems lacking and this is the bottleneck, we can optimize.